tensorboard/data/server/data_compat.rs

```diff
@@ -103,6 +105,51 @@ impl EventValue {
         };
         match *value_box {
             pb::summary::value::Value::Tensor(tp) => Ok(tp),
+            pb::summary::value::Value::Histo(hp) => {
+                // Migrate legacy TF 1.x HistogramProto to TensorProto. The "spec" in summary.proto
+                // says `bucket` and `bucket_limit` are parallel arrays encoding bucket counts and
+                // bucket right edges; the first bucket's left edge is assumed to be -DBL_MAX and
+                // subsequent left edges are defined as the right edge of the preceeding bucket.
+                //
+                // Our conversion logic in data_compat.py however disobeys this and instead sets the
+                // leftmost and rightmost edges to the `min` and `max` values, respectively. This
+                // will result in the outermost buckets having left edge > right edge if they were
+                // originally empty. Apparently the histogram visualization can't handle buckets
+                // extending to -/+ DBL_MAX, but can handle buckets of negative width?
+                //
+                // For consistency with the status quo, we replicate this questionable logic here.
+                let num_buckets = hp.bucket.len();
+                let bucket_edges = || hp.bucket_limit.iter().take(num_buckets - 1).copied();
+                let bucket_lefts = iter::once(hp.min).chain(bucket_edges());
+                let bucket_rights = bucket_edges().chain(iter::once(hp.max));
+                let bucket_counts = hp.bucket.iter().copied();
+                let tensor_content = bucket_lefts
+                    .zip(bucket_rights)
+                    .zip(bucket_counts)
+                    .flat_map(|((l, r), v)| vec![l, r, v])
+                    .map(f64::to_le_bytes)
+                    .collect::<Vec<_>>()
+                    .concat()
+                    .into();
+                Ok(pb::TensorProto {
+                    dtype: pb::DataType::DtDouble.into(),
+                    tensor_shape: Some(pb::TensorShapeProto {
+                        dim: vec![
+                            pb::tensor_shape_proto::Dim {
+                                size: num_buckets as i64,
+                                ..Default::default()
+                            },
+                            pb::tensor_shape_proto::Dim {
+                                size: 3,
+                                ..Default::default()
+                            },
+                        ],
+                        ..Default::default()
+                    }),
```

Comment:
Optional: this is just `tensor_shape(&[num_buckets as i64, 3])` if you
promote `fn tensor_shape` from a test helper to a real helper.

Reply:
Sure, sounds good. Appreciate the awareness of the yak density and happy
that you feel comfortable calling that out.